pkg/operator/operator.go

```diff
@@ -105,24 +105,27 @@ func (o *Operator) Run() error {
 		o.volumeProvisioner,
 		serverVersion.GitVersion,
 	)
-	go pc.Run(wait.NeverStop)
+	go pc.Run(stopChan)
 
 	// watch for changes to the rook clusters
-	o.clusterMgr.Manage()
-	return nil
-}
+	o.clusterController.Watch(v1.NamespaceAll, stopChan)
 
-func (o *Operator) initResources() error {
-	httpCli, err := kit.NewHTTPClient(k8sutil.CustomResourceGroup)
-	if err != nil {
-		return fmt.Errorf("failed to get tpr client. %+v", err)
+	for {
+		select {
+		case <-signalChan:
+			logger.Infof("shutdown signal received, exiting...")
+			close(stopChan)
+			return nil
+		}
 	}
-	o.context.KubeHTTPCli = httpCli.Client
+}
 
-	err = kit.CreateCustomResources(o.context.KubeContext, o.resources)
-	if err != nil {
-		return fmt.Errorf("failed to create TPR. %+v", err)
+func (o *Operator) initResources() error {
+	for _, resource := range o.resources {
+		err := kit.CreateCustomResource(resource, o.context.Clientset)
```

Comment:
can we create all custom resources, then wait for all of them in parallel? At least with TPRs they take a while so didn't want to serialize creation and waiting. If CRDs are faster to create it's not a big deal though.